packages/x-tree-view/src/TreeItem/TreeItem.tsx

```diff
@@ -358,17 +358,17 @@ export const TreeItem = React.forwardRef(function TreeItem(
   });
   const icon = Icon ? <Icon {...iconProps} /> : null;
 
-  let ariaSelected;
-  if (multiSelect) {
-    ariaSelected = selected;
-  } else if (selected) {
-    /* single-selection trees unset aria-selected on un-selected items.
-     *
-     * If the tree does not support multiple selection, aria-selected
-     * is set to true for the selected item and it is not present on any other item in the tree.
-     * Source: https://www.w3.org/WAI/ARIA/apg/patterns/treeview/
-     */
+  // https://www.w3.org/WAI/ARIA/apg/patterns/treeview/
+  let ariaSelected: boolean | undefined;
+  if (selected) {
+    // - each selected node has aria-selected set to true.
     ariaSelected = true;
+  } else if (disableSelection || disabled) {
+    // - if the tree contains nodes that are not selectable, aria-selected is not present on those nodes.
+    ariaSelected = undefined;
```

Comment:
They mentioned that "without the aria-selected=“false”, NVDA doesn't render the information, so we don't know if the item is selected or not."
I'm not sure if NVDA does that only when there's at least one selected, but I'm wondering if, in order to "display" the items, it should always be `false` even when there's no selection or selection is disabled.

Reply:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-selected states for Grid that setting aria-selected="false" on a focusable gridcell indicates the cell is selectable.
I believe the same logic applies to Tree items, so `ariaSelected=undefined` is the way to go indeed.